src/RelayProvider.ts

```diff
@@ -0,0 +1,278 @@
+         * automatically estimated (correctly) by the RelayClient.
+         * The reason why we added this new forceGas field is because at this point, by reading the gas field,
+         * we cannot differentiate between a user-entered gas value from the auto-calculated (and a very poor calculation)
+         * value that comes from the original provider
+         */
+
+        const fullEnvelopingRequest: UserDefinedEnvelopingRequest = {
+            relayData: {
+                ...envelopingRequest.relayData,
+                callForwarder: callForwarderValue,
+            }, 
+            request: {
+                ...envelopingRequest.request,
+                relayHub: relayHubValue,
+                gas: gasToSend,
+            }
+        }
+
+        const fullRequestConfig: RequestConfig = {
+            ...requestConfig,
+            onlyPreferredRelays: onlyPreferredRelaysValue
+        }
+
+        const relayingResult = await this.executeRelayTransaction(fullEnvelopingRequest, fullRequestConfig);
+
+        if(relayingResult.receipt){
+            const { relayRevertedOnRecipient, reason } = this._getRelayStatus(relayingResult.receipt);
+            const { transactionHash } = relayingResult.receipt;
+
+            if (relayRevertedOnRecipient) {
+                throw new Error(
+                        `Transaction Relayed but reverted on recipient - TxHash: ${transactionHash} , Reason: ${reason ?? 'unknown'}`
+                );
+            }
+        }
+
+        return relayingResult;
+    }
+
+    override async listAccounts() {
+        let accountsList = await this.jsonRpcProvider.listAccounts();
+
+        if (accountsList && Array.isArray(accountsList)) {
+            const { chainId } = await this.getNetwork();
+            const accountManager = new AccountManager(this, chainId);
+
+            const ephemeralAccounts =
+                accountManager.getAccounts();
+
+            accountsList = accountsList.concat(ephemeralAccounts);
+        }
+
+        return accountsList;
+    }
+
+    override send(method: string, params: Array<Record<string, unknown>>): Promise<unknown> {
+        if (this._useEnveloping(method, params)) {
+            const { requestConfig, envelopingTx } = params[0] as { requestConfig: RequestConfig, envelopingTx: UserDefinedEnvelopingRequest }
+            const { request } = envelopingTx;
+            if (method === 'eth_sendTransaction') {
+
+                if (!request.to) {
+                    throw new Error(
+                        'Relay Provider cannot relay contract deployment transactions. Add {from: accountWithRBTC, useEnveloping: false}.'
+                    );
+                }
+
+                return this._ethSendTransaction(envelopingTx, requestConfig);
+            }
+
+            if (method === 'eth_accounts') {
+                return this.listAccounts();
+            }
+
+            throw new Error(`Rif Relay unsupported method: ${ method }`);
```

Comment:
By having this throw Error, are we not disabling methods than can executed without the need of enveloping or any change of behavior?

Reply:
you would have to set the useEnveloping flag to true in the request config to get to this error. To not use the custom functionality, one should set useEnveloping to false. At least this is what I understand as the intended use case.

Reply:
@ironFe93 I think that by throwing an error we're assuming that the inner provider isn't able to relay the transaction. Although it makes sense, I'd leave the inner provider to raise the exception instead of raising it directly.

Reply:
True, I have removed this error and leaving it up to the inner provider.